src/cls/rgw/cls_rgw.cc

```diff
@@ -1146,6 +1264,11 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
 		     __func__, escape_str(idx).c_str(), rc);
         return rc;
       }
+      rc = record_duplicate_entry(hctx, idx, entry.key, &new_key_bl,
+                                  header.resharding_in_logrecord(),
+                                  &header.reshardlog_entries);
+      if (rc < 0)
+        return rc;
```

Comment:
can replace this and the `cls_cxx_map_set_val()` call above with `write_entry()`